### PR TITLE
Now `monad_trie_test` `--append` is always to the latest block in db, and bug fix for in memory run

### PR DESCRIPTION
**Pull request by Chen-Yifan**
No functional changes in execution core logic, mainly bug fixes for `monad_trie_test` 
* remove `--block_id` option, now append is always onto the latest block in history
* fix `--in-memory` bug
* fix missed state_nibbles in `--random-read-benchmark`

### CODE REVIEW

File: libs/db/src/monad/mpt/test/monad_trie_test.cpp [hunk 1]
```diff
@@ -506,20 +507,17 @@ int main(int argc, char *argv[])
...
 
             Node::UniquePtr root{};
             if (append) {
                 root.reset(read_node_blocking(
                     io.storage_pool(), aux.get_latest_root_offset()));
             }
-            if (block_id == INVALID_BLOCK_ID) {
-                block_id = aux.db_history_max_version() + 1;
-            }
+            auto block_id = in_memory ? 0 : aux.db_history_max_version() + 1;
```
Comment (ned14) [inline libs/db/src/monad/mpt/test/monad_trie_test.cpp]: Brackets

File: libs/db/src/monad/mpt/test/monad_trie_test.cpp [hunk 2]
```diff
@@ -506,20 +507,17 @@ int main(int argc, char *argv[])
                 }
             };
 
-            UpdateAux<> aux{};
+            UpdateAux<> aux = [&] {
```
Comment (ned14) [inline libs/db/src/monad/mpt/test/monad_trie_test.cpp]: Seems a little overwraught

Comment (Chen-Yifan) [reply libs/db/src/monad/mpt/test/monad_trie_test.cpp]: Ideally I'd like to make `UpdateAux::set_io()` private, this is the effort towards it, but `load_db` lambda in `random_read_benchmark` section still relies on it.

Comment (kkuehlz) [reply libs/db/src/monad/mpt/test/monad_trie_test.cpp]: This being in a lambda seems a tad excessive

**Pull request merged**